packages/components/src/box-model-overlay/test/index.tsx

```diff
@@ -0,0 +1,186 @@
+/**
+ * External dependencies
+ */
+import { render, screen, act } from '@testing-library/react';
+
+/**
+ * WordPress dependencies
+ */
+import { createRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import BoxModelOverlay from '../index';
+import type { BoxModelOverlayHandle } from '../index';
+import { SlotFillProvider, Popover } from '../..';
+
+const DEFAULT_SHOW_VALUES = {
+	margin: {
+		top: true,
+		right: true,
+		bottom: true,
+		left: true,
+	},
+	padding: {
+		top: true,
+		right: true,
+		bottom: true,
+		left: true,
+	},
+};
+
+// Mock ResizeObserver since it's not available in JSDOM yet.
+beforeAll( () => {
+	window.ResizeObserver = jest.fn( () => ( {
+		observe: () => {},
+		unobserve: () => {},
+		disconnect: () => {},
+	} ) );
+} );
+afterAll( () => {
+	window.ResizeObserver = undefined;
+} );
+
+it( 'renders the overlay visible with the children prop', () => {
```

Comment:
Honestly, these tests aren't that helpful for this particular component, since JSDOM doesn't render anything visually.

Perhaps we can set up an infrastructure to write unit tests to run in Playwright in the future.